### DISCUSSION TITLE
Dim to previous state

### DISCUSSION THREAD

**Discussion by LTKingChaos** [plamish/xcomfort#19] [Q&A]
Hi! 

Any way I can dim my lights to previous state? All my lights are dimmed to 100% when using HA no matter what state they wore in before

**Comment by plamish** [comment]
Hey,
Sounds like a cool feature. Would be good to have a configuration option to choose desired behavior for dimmed light. Yes, I can work on that.

**Reply by LTKingChaos** [reply]
That's nice.  I have activated memory function in SHC so it's wierd that is not dimming to last state. 

And thanx for a great integration :)

**Comment by plamish** [comment]
@LTKingChaos can you do me a favor and test some small change? I won't have an physical access to my xComfort devices in near future and can't enable memory function on dimmer actuators to test if small change in the code addresses your need.

Here is updated code for light.py:
https://raw.githubusercontent.com/plamish/xcomfort/master/custom_components/xcomfort/light.py

Please upload it to your HA (/config/custom_components/xcomfort/light.py, restart HA and test if memory function works now.

**Reply by LTKingChaos** [reply]
Hi

Still dimms back to 100%. Let me know if you need any info from SHC :)

**Comment by plamish** [comment]
Hmm...does memory function works well when you turn on/off light with Eaton app or from SHC web interface?

**Reply by LTKingChaos** [reply]
There is definitely something strange in SHC.
From the app and from the Web interface it dimms back to previous state, but both reports that it's 100% even if it's not.

**Comment by plamish** [comment]
Then please test again this change in the code:
https://raw.githubusercontent.com/plamish/xcomfort/master/test/light.py

With this change HA will send exactly the same command to dimmer as the app/web interface. Likely the issue with reporting correct state will stay, but at least you should be able to use the memory function

**Reply by LTKingChaos** [reply]
This is something in between :) 
Now it remembers last state, and goes back to the same state as it was. So far so good. 

But now we can't adjust the brightness it in HA since it always says 100%. Even if I manually slide it to e.g 50% nothing happens

**Comment by plamish** [comment]
Can you try new updated code?
https://raw.githubusercontent.com/plamish/xcomfort/master/test/light.py

In this change the integration should work with memory and allow changing brightness. However the light in HA won't be updated immediately after action. It will be updated only when new status for devices is refreshed.

**Reply by LTKingChaos** [reply]
Awsome. Works like a charm :)

**Reply by LTKingChaos** [reply]
Hi Plamish

Are you gonna integrate this as a standard? I just started a new HA and this is not in the newest release
